sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -177,10 +177,6 @@ protected string GetRecordedOptionalVariable(string name, Action<RecordedVariabl
 
             string value = GetOptionalVariable(name);
 
-            var optionsInstance = new RecordedVariableOptions();
-            options?.Invoke(optionsInstance);
-            var sanitizedValue = optionsInstance.Apply(value);
-
             if (!Mode.HasValue)
```

Comment:
@pakrym:  Based on your feedback on the intent of the options, I reordered things a bit to avoid triggering some code that didn't seem to apply when recording wasn't taking place.   Please let me know if I overlooked something and I'll restore the original structure.